### DISCUSSION TITLE
[Error: EACCES: permission denied, mkdir '/cryptpad/data/logs']

### DISCUSSION THREAD

**Discussion by aquatic7** [louislam/dockge#453] [Ask for Help]
### ⚠️ Please verify that this bug has NOT been reported before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### Description

When i try to deploy Cryptpad, i get the following error, which I don't know is a dockge bug for missing persmission, or if it is an issue with the cryptpad yml or if I have to do some permission settings on my own.:

```
cryptpad-cryptpad-1  | > cryptpad@5.7.0 start
cryptpad-cryptpad-1  | > node server.js
cryptpad-cryptpad-1  | 
cryptpad-cryptpad-1  | /cryptpad/lib/log.js:102
cryptpad-cryptpad-1  |             throw err;
cryptpad-cryptpad-1  |             ^
cryptpad-cryptpad-1  | 
cryptpad-cryptpad-1  | [Error: EACCES: permission denied, mkdir '/cryptpad/data/logs'] {
cryptpad-cryptpad-1  |   errno: -13,
cryptpad-cryptpad-1  |   code: 'EACCES',
cryptpad-cryptpad-1  |   syscall: 'mkdir',
cryptpad-cryptpad-1  |   path: '/cryptpad/data/logs'
cryptpad-cryptpad-1  | }
cryptpad-cryptpad-1  | 
cryptpad-cryptpad-1  | Node.js v20.11.0
```

`![cryptpad error](https://github.com/louislam/dockge/assets/18247537/2483fcda-4f4c-48fb-8602-fb3c885b9449)

### 👟 Reproduction steps

Copy paste the docker-compose.yml

Change the IP/domain as required.

Deploy.

### 👀 Expected behavior

It deploy

### 😓 Actual Behavior

Getting the error as described in the description.

### Dockge Version

1.4.2

### 💻 Operating System and Arch

Ubuntu 22.04.4 LTS

### 🌐 Browser

Firefox 123.0 (64-bit)

### 🐋 Docker Version

Docker version 24.0.5, build 24.0.5-0ubuntu1~22.04.1

### 📝 Relevant log output

```shell
cryptpad-cryptpad-1  | > cryptpad@5.7.0 start
cryptpad-cryptpad-1  | > node server.js
cryptpad-cryptpad-1  | 
cryptpad-cryptpad-1  | /cryptpad/lib/log.js:102
cryptpad-cryptpad-1  |             throw err;
cryptpad-cryptpad-1  |             ^
cryptpad-cryptpad-1  | 
cryptpad-cryptpad-1  | [Error: EACCES: permission denied, mkdir '/cryptpad/data/logs'] {
cryptpad-cryptpad-1  |   errno: -13,
cryptpad-cryptpad-1  |   code: 'EACCES',
cryptpad-cryptpad-1  |   syscall: 'mkdir',
cryptpad-cryptpad-1  |   path: '/cryptpad/data/logs'
cryptpad-cryptpad-1  | }
cryptpad-cryptpad-1  | 
cryptpad-cryptpad-1  | Node.js v20.11.0
```

**Comment by HearMeWhisper** [comment]
I'm having the same issue...

**Comment by BurningBOB** [comment]
Yep same here. setting the permissions on the folder to full rw for owner, group, and other fixes it. i even tried setting dockge and/or the container to root and still dosnt create the folders with full permissions. whats odd is that i have other containers that are working fine without full permissions but it seems new containers have this issue now.

**Comment by DrHerbalist** [comment]
The user and group of cryptpad both run as uid 4001 and gid 4001. 

You will need to set the filesystem user permissions in your docker/compose file.

**Comment by aelharda** [comment]
in TrueNAS shell, run `sudo chown -R 4001:4001 /pathofyourdataset/cryptpad/`
